app/mempool/proposal_handler.go

```diff
@@ -0,0 +1,85 @@
+package mempool
+
+import (
+	"fmt"
+
+	abci "github.com/cometbft/cometbft/abci/types"
+
+	"cosmossdk.io/log"
+
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+type (
+	// ProposalHandler wraps ABCI++ PrepareProposal/ProcessProposal for the Mempool.
+	ProposalHandler struct {
+		logger                   log.Logger
+		txDecoder                sdk.TxDecoder
+		Mempool                  *Mempool
+		useCustomProcessProposal bool
+	}
+)
```

Comment:
why wrap in type? why not

```suggestion
// ProposalHandler wraps ABCI++ PrepareProposal/ProcessProposal for the Mempool.
type ProposalHandler struct {
	logger                   log.Logger
	txDecoder                sdk.TxDecoder
	Mempool                  *Mempool
	useCustomProcessProposal bool
}
```

Reply:
ok